CHANGELOG.md

```diff
@@ -1,5 +1,19 @@
 # HSD Release Notes & Changelog
 
+## unreleased
+
+### Node changes
+
+- `FullNode` has new RPC command `compacttree` which reduces the Urkel Tree disk
+usage by deleting historical data. It will keep up to the last 288 blocks worth
+of tree data on disk (7-8 tree intervals) exposing the node to a similar deep
+reorganization vulnerability as a chain-pruning node.
```

Comment:
Great explanation here. 

I wonder if this suggests a want for a mechanism to detect a large re-org and warn the operator / bail out. I understand the likelihood of this occurring on mainnet is very small but if the worst case scenario is feasible maybe it's a nice emergency brake to have. Unless there is already something for that.

Reply:
hsd will warn in the log when a block competitor has been discovered and again when a reorg actually happens. There are also events emitted so a serious node operator or complex application can react immediately. In the event that a reorg goes deeper than the prune depth (2 days / 288 blocks) there's not much warning to be had: the node just crashes, there's no other choice.

Reply:
thanks for clarifying, that seems adequate